design/commands.md

```diff
@@ -0,0 +1,201 @@
+# Commands (Sharing Binaries)
+
+## Context
+
+Currently, there's no clear way to share a binaries (or scripts) across multiple jobs.  Some people have chosen to share via git repositories, yum packages, node modules, and even docker images.  But nothing is standardized and works easily in all containers.
+
+## Status
+
+- 11 September 2017: Proposal submitted
+
+## Proposal
+
+Have a single interface for executing a versioned command (via remote binary, docker image, or habitat package) during a Screwdriver build.  Managed and discoverable through the Screwdriver interface.
+
+This provides a single abstraction for execution of shared code.
+
+```bash
+$ sd_cmd chefdk/knife@12 search node '*:*'
+$ sd_cmd nodejs/vuln_check@2 ./package.json
+```
+
+## Details
+
+### Specification
+
+```yaml
+# Namespace for the command
+namespace: foo
+# Command name itself
+command: bar
+# Description of the command and what it does
+description: |
+  Lorem ipsum dolor sit amet.
+# Major and Minor version number (patch is automatic)
+version: 1.0
+# Format the command is in (see below)
+format: habitat
```

Comment:
Is this supposed to be an array of available formats?
Why are `docker` & `binary` config defined, but `format` doesn't include them?

Reply:
No, the items below are supposed to be an example.  I'll update it.